test/fuzz/gen.ml

```diff
@@ -67,137 +76,177 @@ let expr_available_2_i64_i32 (env : Env.t) =
-  [ pair B.funop_64 (const [ S.Nothing ])
-  ; pair B.i32_trunc_f64 (const [ S.Pop; S.Push (Num_type I32) ])
-  ; pair B.i64_trunc_f64 (const [ S.Pop; S.Push (Num_type I64) ])
-  ; pair B.i32_trunc_sat_f64 (const [ S.Pop; S.Push (Num_type I32) ])
-  ; pair B.i64_trunc_sat_f64 (const [ S.Pop; S.Push (Num_type I64) ])
-  ; pair B.f32_demote_f64 (const [ S.Pop; S.Push (Num_type F32) ])
-  ; pair B.i64_reinterpret_f64 (const [ S.Pop; S.Push (Num_type I64) ])
-  ]
-  @ B.local_set_f64 env @ B.local_tee_f64 env @ B.global_set_f64 env
+let expr_available_1_f64 env : _ condgen =
+  (not_symbolic @@ pair B.i32_trunc_f64 (const [ S.Pop; S.Push (Num_type I32) ]))
+  **::** ( not_symbolic
+         @@ pair B.i64_trunc_f64 (const [ S.Pop; S.Push (Num_type I64) ]) )
+  **::** ( not_symbolic
+         @@ pair B.i32_trunc_sat_f64 (const [ S.Pop; S.Push (Num_type I32) ]) )
+  **::** ( not_symbolic
+         @@ pair B.i64_trunc_sat_f64 (const [ S.Pop; S.Push (Num_type I64) ]) )
+  **::** ( not_symbolic
+         @@ pair B.f32_demote_f64 (const [ S.Pop; S.Push (Num_type F32) ]) )
+  **::** ( not_symbolic
+         @@ pair B.i64_reinterpret_f64 (const [ S.Pop; S.Push (Num_type I64) ])
+         )
+  **::** (not_symbolic @@ pair B.funop_trunc_64 (const [ S.Nothing ]))
+  **:: ( [ pair B.funop_64 (const [ S.Nothing ]) ]
+       @ B.local_set_f64 env @ B.local_tee_f64 env @ B.global_set_f64 env )
 
-let expr_available_2_f64 =
-  [ pair B.fbinop_64 (const [ S.Pop ])
-  ; pair B.frelop_64 (const [ S.Pop; S.Pop; S.Push (Num_type I32) ])
-  ]
+let expr_available_2_f64 : _ condgen =
+  (not_symbolic @@ pair B.fbinop_64 (const [ S.Pop ]))
+  **:: [ pair B.fbinop_64 (const [ S.Pop ])
+       ; pair B.frelop_64 (const [ S.Pop; S.Pop; S.Push (Num_type I32) ])
+       ]
 
 (* let expr_available_3_f64 = [] *)
 
 let if_else expr ~locals ~stack env =
-  match stack with
-  | Num_type I32 :: _stack -> begin
-    let* rt = list B.val_type in
-    let* pt = B.stack_prefix (List.tl stack) in
-    let typ =
-      Bt_raw (None, (List.rev_map (fun t -> (None, t)) pt, List.rev rt))
-    in
-    let id = Env.add_block env typ Env.Block in
-    (* same behavior as block *)
-    let old_fuel = env.Env.fuel in
-    env.fuel <- old_fuel / 2;
-    let* expr_then = expr ~block_type:typ ~stack:pt ~locals env in
-    env.fuel <- old_fuel / 2;
-    let* expr_else = expr ~block_type:typ ~stack:pt ~locals env in
-    env.fuel <- old_fuel / 2;
-    Env.remove_block env;
-    let+ instr = const @@ If_else (Some id, Some typ, expr_then, expr_else)
-    and+ pt_descr = const @@ (S.Pop :: List.map (fun _ -> S.Pop) pt)
-    and+ rt_descr = const @@ List.rev_map (fun t -> S.Push t) rt in
-    (instr, pt_descr @ rt_descr)
-  end
-  | _ -> assert false
+  Cond.with_unwrapping_configurator (fun configurator ->
+      let open Crowbar.Syntax in
+      match stack with
+      | Num_type I32 :: _stack -> begin
+        let* rt = list B.val_type in
+        let* pt = B.stack_prefix (List.tl stack) in
```

Comment:
This is not because of this PR but the code would be better written like this:

```ocaml
      match stack with
      | Num_type I32 :: stack -> begin
        let* rt = list B.val_type in
        let* pt = B.stack_prefix stack in
```